src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -195,6 +196,10 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return str_getcsv($env);
         }
 
+        if ('nullable' === $prefix) {
+            return '' === $env ? null : $env;
```

Comment:
What if the env var is not defined at all? To me, not providing the env var means `null`.

Currently, you can already get a `null` behavior for `FOO` env var by setting a `env(FOO)` parameter:

```yml
parameters:
    env(FOO): null
```

so `null` is used as default value if  `FOO` env var is not set.

Don't we just need this allowing an env var not to be set to get `null`, without defining a parameter as default?

Reply:
i think we should avoid

```
env(DEFAULT_NULL): null
%env(DEFAULT_NULL)%
```

competing with

```
%env(nullable:RUNTIME_ARBITRARY_NULL)%
```

to take this further we could also deprecate setting null defaults, in favor of always using `nullable:` (related to #27808)

Reply:
```
%env(DEFAULT_ENV)%
%env(nullable:UNKNOWN_ENV)%
```

Here, my worry is `RUNTIME_ENV=` can only be nulled by removing it or move it to config.

Currently there's only one source where null can come from, and it's: `env(DEFAULT): ~`

Adding `nullable` creates 2 scenarios. Hence my suggestion to deprecate `env(DEFAULT): ~` in favor of

```
RUNTIME=
env(DEFAULT): ""

%env(nullable:DEFAULT)%
%env(nullable:RUNTIME)%
```

IMHO that's the easiest to reason about.

Reply:
When the env var is not defined, we already added the "default" processor in #28976. I think this processor should do one thing, turn empty string to null - what it does currently - and "default" covers the other use cases.
I also don't agree deprecating `env(FOO): ~` is a good idea: we shouldn't deprecate for the sake of it.

Reply:
Note that "default" has not been released yet so we can also alter its behavior.
But each processor should have one responsibility - we shouldn't have many processors do the same in slightly different ways.